x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/DatafeedJob.java

```diff
@@ -248,11 +248,16 @@ private void checkForMissingDataIfNecessary() {
                 }
 
                 if (lastDataCheckAnnotationWithId == null) {
-                    lastDataCheckAnnotationWithId = annotationPersister.persistAnnotation(null, annotation);
+                    Optional<Tuple<String, Annotation>> maybeAnnotation = annotationPersister.persistAnnotation(null, annotation);
```

Comment:
This `if` block can be simplified so that there is a single call to `persistAnnotation`.

It can be:

```
String annotationId = null;
if (lastDataCheckAnnotationWithId != null) {
  annotationId = lastDataCheckAnnotationWithId.v1();
  Annotation updatedAnnotation = updateAnnotation(annotation);
}
try {
  lastDataCheckAnnotationWithId = annotationPersister.persistAnnotation(annotationId, annotation);
} catch (ElasticsearchException e) {
  ...
}
```